code/modules/atmospherics/machinery/atmosmachinery.dm

```diff
@@ -460,7 +460,7 @@
 		add_atom_colour(obj_color, FIXED_COLOUR_PRIORITY)
 		pipe_color = obj_color
 	update_name()
-	set_piping_layer(set_layer)
+	set_piping_layer(set_layer || piping_layer || PIPING_LAYER_DEFAULT)
```

Comment:
No need to do both piping_layer and PIPING_LAYER_DEFAULT since they are both the same.
Also i believe it would be better to have on_construction(obj_color, set_layer = PIPING_LAYER_DEFAULT)

Reply:
yeah i guess that makes sense